### ISSUE TITLE
besside-ng crashes with "wi_read(): No child processes"

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1558]
**Reported by chronos00 on 1 Feb 2015 03:45 UTC**

I've ran besside-ng countless times with many onboard and external WiFi adapters, and it always ran flawlessly.

This time however, it consistently crashes with my new laptop's WiFi (Intel's Wireless 7260). Sometimes it takes longer, sometimes shorter, but always less than 10 seconds.

This is a typical run, after which you can see it crashes and returns to prompt.
```
chronos@SGZB:~/Downloads/aircrack-ng$ sudo besside-ng -v wlan2mon
[mac 7c:7a:91:xx:xx:xx
[00:14:40](00:14:40]) Let's ride
[Resuming from besside.log
[00:14:40](00:14:40]) Found AP 08:86:3b:13:8d:4e [chan 0 crypto WPA dbm 0
[00:14:40](AP1103]) Appending to wpa.cap
[Appending to wep.cap
[00:14:40](00:14:40]) Logging to besside.log
[Found AP f4:ec:38:a8:18:8c [WiFi-VanGogh](00:14:40]) chan 2 crypto WPA dbm -90
[Found AP 4c:5e:0c:68:30:f5 [TorrePlaza](00:14:41]) chan 4 crypto WPA dbm -88
[Found AP 14:60:80:b0:f6:a7 [AP 1003](00:14:41]) chan 6 crypto WPA dbm -79
[Found AP 14:60:80:b0:c7:77 [AP 1004](00:14:41]) chan 6 crypto WPA dbm -80
[Found AP 00:0c:41:f5:bd:b8 [AP903](00:14:41]) chan 6 crypto WPA dbm -67
[Found AP 00:25:9c:2f:44:75 [thelmapunta](00:14:41]) chan 6 crypto WPA dbm -84
[Found AP 14:60:80:b0:fa:5d [apto1104](00:14:41]) chan 6 crypto WPA dbm -79
[Found AP 14:60:80:b0:fa:71 [AP 403](00:14:41]) chan 7 crypto WPA dbm -83
[Found AP 14:60:80:b0:fb:f7 [apto703](00:14:42]) chan 8 crypto WPA dbm -84
[Found AP 14:60:80:b0:f9:ea [apto1103](00:14:42]) chan 9 crypto WPA dbm -63
[Found AP a0:f3:c1:f6:c0:aa [Garden 901](00:14:42]) chan 8 crypto WPA dbm -84
[Found AP 54:22:f8:93:7d:5a [Tito](00:14:42]) chan 9 crypto WPA dbm -88
[Found AP f4:ec:38:d0:e0:e8 [AP704](00:14:42]) chan 11 crypto WPA dbm -82
[Found AP 00:11:3b:17:65:33 [PTP_YACHT](00:14:42]) chan 11 crypto WEP dbm -79
[Found AP 90:f6:52:29:f1:32 [AP1104](00:14:43]) chan 11 crypto WPA dbm -84
[Found AP 2c:54:2d:38:67:0b [balanzas](00:14:43]) chan 11 crypto WPA dbm -89
[Found client for network [balanzas](00:14:43]) c0:4a:00:52:e6:c8
[Found client for network [AP1103](00:14:43]) 28:cf:da:d4:77:78
besside-ng: wi_read(): No child processes
chronos@SGZB:~/Downloads/aircrack-ng$
```
----
'''besside-ng version'''
```
besside-ng -v
Gimme an interface name dude

  Besside-ng 1.2 rc1 aircrack-ng/aircrack-ng@b2098a4c5f28904131e965f69a6233122fe3a964 - (C) 2010 Andrea Bittau
```
----
Laptop's make and model:
```
ASUS Zenbook UX301LA
```
----
'''lshw output (regarding my WiFi adapter):'''
```
        *-network
             description: Wireless interface
             product: Wireless 7260
             vendor: Intel Corporation
             physical id: 0
             bus info: pci@0000:02:00.0
             logical name: wlan2
             version: 6b
             serial: 7c:7a:91:xx:xx:xx (hiden for privacy)
             width: 64 bits
             clock: 33MHz
             capabilities: pm msi pciexpress bus_master cap_list ethernet physical wireless
             configuration: broadcast=yes driver=iwlwifi driverversion=3.16.0-29-generic firmware=25.228.9.0 ip=192.168.1.44 latency=0 link=yes multicast=yes wireless=IEEE 802.11abgn
             resources: irq:64 memory:f7c00000-f7c01fff
```

**Comment by aircrack-ng** [comment]
**Comment by chronos00 on 1 Feb 2015 03:45 UTC**

By the way, Just to make it clear, I am using the latest code in trunk, dated January 1st, 2015.

Does anybody else see this behavior?

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 1 Feb 2015 03:45 UTC**

Well, me too,
besside-ng chrashes directly after startup with
besside-ng: wi_read(): No child processes
next run ist ok and all works as expected
only every first start of besside-ng crashes

**Comment by aircrack-ng** [comment]
**Comment by misterx on 1 Feb 2015 03:45 UTC**

I can't get to reproduce the bug. What Linux distribution are you using? Kernel version?

**Comment by aircrack-ng** [comment]
**Comment by chronos00 on 1 Feb 2015 03:45 UTC**

I am using Ubuntu 14.10.

Kernel at this time: (besside-ng keeps crashing at this time, with this kernel)
```
chronos@SGZB:~$ uname -a
Linux SGZB 3.16.0-37-generic #51-Ubuntu SMP Tue May 5 13:45:59 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```


I have stopped all network related processes, of course, with:
```
sudo service avahi-daemon stop ;
sudo service network-manager stop ;
sudo killall wpa_supplicant ;
sudo airmon-ng start wlan0
```

**Comment by aircrack-ng** [comment]
**Comment by misterx on 1 Feb 2015 03:45 UTC**

With what error(s)?

**Comment by aircrack-ng** [comment]
**Comment by strasharo2000 on 1 Feb 2015 03:45 UTC**

For me besside-ng is constantly crashing with the following error:
[tmp](root@pie)# aircrack-ng/src/besside-ng -vvv wlan0mon
[mac 00:c0:ca:72:6c:61
[10:22:58](10:22:58]) Let's ride
[Resuming from besside.log
[10:22:58](10:22:58]) Found AP 00:24:d1:83:58:cc [chan 0 crypto WPA dbm 0
[10:22:58](blizoo2]) Appending to wpa.cap
[Appending to wep.cap
[10:22:58](10:22:58]) Logging to besside.log
[Found AP b0:48:7a:e9:59:c6 [Panch0thePuss](10:23:02]) chan 2 crypto WPA dbm -95
[Found AP 4c:28:02:10:0e:04 [KOP](10:23:02]) chan 1 crypto WPA dbm -25
[Found AP c8:b3:73:39:76:f7 [Agrobio](10:23:03]) chan 1 crypto WPA dbm -86
[Found AP b4:75:0e:85:04:0e [metalista](10:23:03]) chan 1 crypto WPA dbm -77
[Found AP 10:c6:1f:04:4f:e0 [VIVACOM](10:23:03]) chan 3 crypto WPA dbm -70
'''besside-ng: wi_read(): No child processes'''
[tmp](root@pie)#

I'm running the latest version from svn.

Platform is Archlinux running on Raspberry Pi:
Linux pie 4.1.4-1-ARCH #1 PREEMPT Thu Aug 6 21:07:48 MDT 2015 armv6l GNU/Linux

The wireless adapter is AWUS036NHA. 
Everything else is working okay there.

P.S. I tried running besside-ng on a Kali VM using the same adapter and it doesn't crash there.

**Comment by aircrack-ng** [comment]
**Comment by strasharo2000 on 1 Feb 2015 03:45 UTC**

Same happens on a KVM VM with Kali 2.0 and BeagleBone Black running Archlinux. It looks that the only distribution so far, where the tool worked flawlessly is the first version of Kali.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 1 Feb 2015 03:45 UTC**

#1627 is similar, if not a duplicate.

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 1 Feb 2015 03:45 UTC**

Yes, indeed. It looks that the issue is somehow specific/related to the ath9k_htc driver. Recently I bought an Alfa AWUS036H V5 (Realtek Semiconductor Corp. RTL8187/rtl8187) and TP-Link TL-WN722N (Atheros Communications, Inc. AR9271 802.11n/driver ath9k_htc). I'm experiencing the same issue with the newly bought TP-Link as I do with my AWUS036NHA (Atheros Communications, Inc. AR9271 802.11n/driver ath9k_htc), while the Realtek AWUS036H works without any issues. If I might be able to assist with the troubleshooting of this issue, just let me know.

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 1 Feb 2015 03:45 UTC**

This bug is not specific to ath9k_htc driver, maybe ath9k in general? I get the very same bug on a TP-Link TL-WR1043N/ND v1 Router running OpenWRT Chaos Calmer 15.05 (Kernel 3.18.20), the ath9k_htc driver is not installed.

**Comment by aircrack-ng** [comment]
**Comment by strasharo2000 on 1 Feb 2015 03:45 UTC**

I think this might be somehow related:

https://wigle.net/phpbb/viewtopic.php?t=2261#p8416

**Comment by aircrack-ng** [comment]
**Comment by strasharo2000 on 1 Feb 2015 03:45 UTC**

There's a confirmed working fix for the issue over here, which hasn't been accepted yet:
https://github.com/aircrack-ng/aircrack-ng/pull/66

**Comment by aircrack-ng** [comment]
**Comment by misterx on 1 Feb 2015 03:45 UTC**

Fixed: https://github.com/aircrack-ng/aircrack-ng/pull/115

Was committed in aircrack-ng/aircrack-ng@24946aa63db3d6f38671d90ca584c2885177ffd6

**Comment by aircrack-ng** [comment]
**Comment by misterx on 1 Feb 2015 03:45 UTC**

Fixed in #2854
